File: DesignPatterns/builder.py
class Application:
    def __init__(self):
        self.configA=0
        self.configB=0
        self.configC=0
    
    def add_all_config(self):
        return self.configA+self.configB+self.configC
    

class ApplicationBuilder:
    def __init__(self):
        self.application=Application()

    def set_configA(self,configA):
        self.application.configA=configA
        return self
    
    def set_configB(self,configB):
        self.application.configB=configB
        return self
    
    def set_configC(self,configC):
        self.application.configC=configC
        return self
    
    def build(self):
        return self.application
    

if __name__ == "__main__":
    builder=ApplicationBuilder()
    application=builder.set_configA(3).set_configB(2).set_configC(3).build()
    print(application.add_all_config())  # Output: 6
    

File: dotnet/ConsoleApp1/ConsoleApp1/Creational/Builder/Builder.cs
//Builder design pattern is used when you want to take away logic of building a complex object and give it to some other objext.


ApplicationBuilder builder = new ApplicationBuilder();
Application appl=builder.SetConfigA(1).SetConfigB(2).SetConfigC(3).Build();
Console.WriteLine(appl.AddAllConfigs());


class Application
{
    public int ConfigA { get; set; }
    public int ConfigB { get; set; }
    public int ConfigC { get; set; }
    
    public int AddAllConfigs()
    {
        return ConfigA+ConfigB+ConfigC;
    }

}

class ApplicationBuilder
{
    private Application application;

    public ApplicationBuilder()
    {
        application = new Application();
    }
    public ApplicationBuilder SetConfigA(int configA)
    {
        application.ConfigA = configA;
        return this;
    }
    public ApplicationBuilder SetConfigB(int configB)
    {
        application.ConfigB = configB;
        return this;
    }

    public ApplicationBuilder SetConfigC(int configC)
    {
        application.ConfigC = configC;
        return this;
    }
    public Application Build()
    {
        return application;
    }

}